packages/zeplin-extension-style-kit/values/gradient/linearGradient.js

```diff
@@ -0,0 +1,127 @@
+import LinearColorStop from "./linearColorStop";
+
+class LinearGradient {
+    // eslint-disable-next-line max-params
+    constructor(colorStops, from, to, width, height) {
+        const x1 = from.x * width;
+        const y1 = from.y * height;
+        const x2 = to.x * width;
+        const y2 = to.y * height;
+
+        this.angle = (360 - Math.round(Math.atan2(x1 - x2, y1 - y2) * 180 / Math.PI)) % 360;
+
+        let pFirst;
+        let pLast;
+        let pStart;
+        let pEnd;
+
+        // Find the slope of the gradient line
+        const m1 = (y2 - y1) / (x2 - x1);
+
+        if (m1 === 0) {
+            // If the slope is 0, gradient line is horizontal
+            pFirst = x1;
+            pLast = x2;
+            pStart = 0;
+            pEnd = width;
+        } else if (!Number.isFinite(m1)) {
+            // If the slope is ±Infinity, gradient line is vertical
+            pFirst = y1;
+            pLast = y2;
+            pStart = 0;
+            pEnd = height;
+        } else {
+            pFirst = x1;
+            pLast = x2;
+
+            // Find c1 such that y = m1 * x + c1, by applying one of the positions
+            const c1 = y1 - m1 * x1;
+
+            // The slope of a perpendicular line
+            const m2 = -1 / m1;
+
+            // Starting and ending points
+            let x0;
+            let y0;
+            let xe;
+            let ye;
+
+            if (m1 > 0) {
+                // if the slope is positive, starting point is top-left corner (0, 0)
+                // and ending point is bottom-right corner (w, h)
+                x0 = 0;
+                y0 = 0;
+                xe = width;
+                ye = height;
+            } else {
+                // if the slope is negative, starting point is bottom-left corner (0, h)
+                // and ending point is top-right corner (w, 0)
+                x0 = 0;
+                y0 = height;
+                xe = width;
+                ye = 0;
+            }
+
+            // c2 equation constant of the line crossing from starting point
+            // and perpendicular to the gradient line
+            const c2 = y0 - m2 * x0;
+
+            // x position of the intersection point of gradient line
+            // and the perpendecular line crossing from starting point
+            pStart = (c2 - c1) / (m1 - m2);
+
+            // c3 equation constant of the line crossing from starting point
+            // and perpendicular to the gradient line
+            const c3 = ye - m2 * xe;
+
+            // x position of the intersection point of gradient line
+            // and the perpendecular line crossing from ending point
+            pEnd = (c3 - c1) / (m1 - m2);
+        }
+
+        this.colorStops = colorStops.map(cs => new LinearColorStop(cs, pFirst, pLast, pStart, pEnd));
```

Comment:
Instead of just calculating intersection points with gradient line and passing those points to `LinearColorStop` to calculate its position, it would be better to calculate it in here and not leak calculating position logic to `LinearColorStop`s constructor, wdyt? @merih

Reply:
We do the same inner calculation for both radial and linear gradients. Do you it'd be better to duplicate that in both of them?